Hardcoded warnings level

Hi! I use your project with CMake [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html) feature because the library is not available in all Linux distibutions / vcpkg on Windows.
But you have hardcoded warnings level for [MSVC](https://github.com/stp/stp/blob/81850c0d6973bf5fb7d283ab4037385930290be2/CMakeLists.txt#L239) and [GCC](https://github.com/stp/stp/blob/master/CMakeLists.txt#L308).

Because of it I have the following warnings on GCC 11.1 in your project:

<details>
  <summary>GCC</summary>

```
build/_deps/stp-src/lib/extlib-abc/vecPtr.h|747 col 13| warning: cast between incompatible function types from «int (*)()» to «int (*)(const void*, const void*)» [-Wcast-function-type]
   747 |             (int (*)(const void *, const void *)) Vec_PtrSortCompare );
       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 build/_deps/stp-src/lib/extlib-abc/vecPtr.h: In function: «void Vec_PtrUniqify(Vec_Ptr_t*, int (*)())»:
build/_deps/stp-src/lib/extlib-abc/vecPtr.h|767 col 13| warning: cast between incompatible function types from «int (*)()» to «int (*)(const void*, const void*)» [-Wcast-function-type]
   767 |             (int (*)(const void *, const void *)) Vec_PtrSortCompare );
       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 In file included from build/_deps/stp-src/include/stp/ToSat/AIG/BBNodeManagerAIG.h:34,
 build/_deps/stp-src/lib/extlib-abc/cnf_short.h: На глобальном уровне:
build/_deps/stp-src/lib/extlib-abc/cnf_short.h|90 col 29| warning: ISO C ++ disallow zero size array «pFanins» [-Wpedantic]
    90 |     int             pFanins[0];      // the fanins (followed by the truth table)
       |                             ^
 build/_deps/stp-src/lib/Simplifier/Simplifier.cpp: In function:-члене «stp::ASTNode stp::Simplifier::simplify_term_switch(const stp::ASTNode&, stp::ASTNode&, stp::ASTNode&, stp::ASTNodeMap*, stp::Kind, unsigned int)»:
build/_deps/stp-src/lib/Simplifier/Simplifier.cpp|2037 col 5| warning: this operator may fallthrough [-Wimplicit-fallthrough=]
  2037 |     }
       |     ^
build/_deps/stp-src/lib/Simplifier/Simplifier.cpp|2041 col 5| here:
  2041 |     case BVPLUS:
       |     ^~~~
 [33/125] Building CXX object _deps/stp-build/lib/Simplifier/CMakeFiles/simplifier.dir/DifficultyScore.cpp.o
 build/_deps/stp-src/lib/Simplifier/DifficultyScore.cpp: In function: «long int stp::eval(const stp::ASTNode&)»:
build/_deps/stp-src/lib/Simplifier/DifficultyScore.cpp|57 col 22| warning: comparison of integer expressions of different signedness: «int» and «unsigned int» [-Wsign-compare]
    57 |     for (int i =1; i < b.GetValueWidth();i++)
       |                    ~~^~~~~~~~~~~~~~~~~~~
```
  
</details>

On MSVC I have the following warnings in your project:

<details>
  <summary>MSVC</summary>

```  
[build] [10/124   0% :: 0.145] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigFanout.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [11/124   1% :: 0.148] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigMffc.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [12/124   2% :: 0.155] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigObj.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [13/124   3% :: 0.158] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigCheck.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [14/124   4% :: 0.161] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigOper.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [15/124   4% :: 0.163] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigDfs.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [16/124   5% :: 0.174] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigMan.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [17/124   6% :: 0.177] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigPart.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-src\lib\extlib-abc\aig\aig\aigPart.c(295): warning C4312: 'type cast': conversion from 'long' to 'Aig_Obj_t *' of greater size
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-src\lib\extlib-abc\aig\aig\aigPart.c(297): warning C4312: 'type cast': conversion from 'long' to 'Aig_Obj_t *' of greater size
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-src\lib\extlib-abc\aig\aig\aigPart.c(322): warning C4311: 'type cast': pointer truncation from 'Aig_Obj_t *' to 'long'
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-src\lib\extlib-abc\aig\aig\aigPart.c(334): warning C4311: 'type cast': pointer truncation from 'Aig_Obj_t *' to 'long'
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-src\lib\extlib-abc\aig\aig\aigPart.c(749): warning C4311: 'type cast': pointer truncation from 'Aig_Obj_t *' to 'long'
[build] [18/124   7% :: 0.180] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigMem.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [19/124   8% :: 0.184] Generating ASTKind.cpp, ../../include/stp/AST/ASTKind.h
[build] [20/124   8% :: 0.319] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigRepr.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [21/124   9% :: 0.323] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigScl.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-src\lib\extlib-abc\aig\aig\aigScl.c(180): warning C4312: 'type cast': conversion from 'long' to 'Aig_Obj_t *' of greater size
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-src\lib\extlib-abc\aig\aig\aigScl.c(183): warning C4311: 'type cast': pointer truncation from 'Aig_Obj_t *' to 'long'
[build] [22/124  10% :: 0.326] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigRet.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [23/124  11% :: 0.336] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigOrder.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [24/124  12% :: 0.348] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigSeq.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [25/124  12% :: 0.386] Building CXX object _deps\stp-build\lib\AST\CMakeFiles\AST.dir\ASTKind.cpp.obj
[build] [26/124  13% :: 0.805] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\dar\darMan.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [27/124  14% :: 0.808] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\dar\darRefact.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [28/124  15% :: 0.810] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\dar\darScript.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [29/124  16% :: 0.813] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\dar\darPrec.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [74/124  52% :: 14.794] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\kit\kitGraph.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-src\lib\extlib-abc\aig\kit\kitGraph.c(342): warning C4312: 'type cast': conversion from 'long' to 'void *' of greater size
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-src\lib\extlib-abc\aig\kit\kitGraph.c(347): warning C4311: 'type cast': pointer truncation from 'void *' to 'long'
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-src\lib\extlib-abc\aig\kit\kitGraph.c(348): warning C4311: 'type cast': pointer truncation from 'void *' to 'long'
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-src\lib\extlib-abc\aig\kit\kitGraph.c(352): warning C4312: 'type cast': conversion from 'long' to 'void *' of greater size
[build] [75/124  53% :: 14.989] Generating parsesmt2.tab.cpp, parsesmt2.tab.h, lexsmt2.cpp
[build] D:/Work/FortifyIQ-win-Ninja-Debug/_deps/stp-src/lib/Parser/smt2.y: warning: 1 reduce/reduce conflict [-Wconflicts-rr]
[build] D:/Work/FortifyIQ-win-Ninja-Debug/_deps/stp-src/lib/Parser/smt2.y: note: rerun with option '-Wcounterexamples' to generate conflict counterexamples
[build] [76/124  54% :: 15.140] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\kit\kitAig.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [77/124  54% :: 15.275] Building CXX object _deps\stp-build\lib\Printer\CMakeFiles\printer.dir\dotPrinter.cpp.obj
[build] [78/124  55% :: 15.287] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigTsim.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [79/124  56% :: 15.401] Building CXX object _deps\stp-build\lib\Printer\CMakeFiles\printer.dir\GDLPrinter.cpp.obj
[build] [80/124  57% :: 15.454] Building CXX object _deps\stp-build\lib\Printer\CMakeFiles\printer.dir\LispPrinter.cpp.obj
[build] [81/124  58% :: 15.473] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigTime.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [82/124  58% :: 15.542] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigShow.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [83/124  59% :: 15.581] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigTable.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [84/124  60% :: 15.681] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigTiming.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [85/124  61% :: 15.686] Building CXX object _deps\stp-build\lib\Printer\CMakeFiles\printer.dir\SMTLIBPrinter.cpp.obj
[build] [86/124  62% :: 15.693] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigTruth.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [87/124  62% :: 15.717] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\cnf\cnfMan.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [88/124  63% :: 15.789] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigWin.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [90/124  65% :: 15.890] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\cnf\cnfCut.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [91/124  66% :: 15.929] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\cnf\cnfCore.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [92/124  66% :: 15.955] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\cnf\cnfData.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [93/124  67% :: 16.011] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\dar\darCore.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [94/124  68% :: 16.017] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\aig\aigUtil.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [95/124  69% :: 16.087] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\dar\darBalance.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [96/124  70% :: 17.304] Building CXX object _deps\stp-build\lib\Interface\CMakeFiles\cinterface.dir\c_interface.cpp.obj
[build] [97/124  70% :: 17.625] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\cnf\cnfWrite.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [98/124  71% :: 17.807] Building CXX object _deps\stp-build\lib\Parser\CMakeFiles\parser.dir\LetMgr.cpp.obj
[build] [99/124  72% :: 18.059] Building CXX object _deps\stp-build\lib\Interface\CMakeFiles\cppinterface.dir\cpp_interface.cpp.obj
[build] [100/124  73% :: 18.233] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\cnf\cnfMap.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [101/124  74% :: 18.373] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\cnf\cnfUtil.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [102/124  75% :: 18.507] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\cnf\cnfPost.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [103/124  75% :: 18.966] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\dar\darCut.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [104/124  76% :: 19.081] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\dar\darData.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [105/124  77% :: 19.120] Building CXX object _deps\stp-build\lib\Parser\CMakeFiles\parser.dir\lexcvc.cpp.obj
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(49): warning C4005: 'INT8_MIN': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexcvc.cpp(318): note: see previous definition of 'INT8_MIN'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(50): warning C4005: 'INT16_MIN': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexcvc.cpp(321): note: see previous definition of 'INT16_MIN'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(51): warning C4005: 'INT32_MIN': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexcvc.cpp(324): note: see previous definition of 'INT32_MIN'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(53): warning C4005: 'INT8_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexcvc.cpp(327): note: see previous definition of 'INT8_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(54): warning C4005: 'INT16_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexcvc.cpp(330): note: see previous definition of 'INT16_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(55): warning C4005: 'INT32_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexcvc.cpp(333): note: see previous definition of 'INT32_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(57): warning C4005: 'UINT8_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexcvc.cpp(336): note: see previous definition of 'UINT8_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(58): warning C4005: 'UINT16_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexcvc.cpp(339): note: see previous definition of 'UINT16_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(59): warning C4005: 'UINT32_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexcvc.cpp(342): note: see previous definition of 'UINT32_MAX'
[build] [106/124  78% :: 19.123] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\dar\darLib.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [107/124  79% :: 19.161] Building CXX object _deps\stp-build\lib\Parser\CMakeFiles\parser.dir\lexsmt.cpp.obj
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(49): warning C4005: 'INT8_MIN': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt.cpp(318): note: see previous definition of 'INT8_MIN'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(50): warning C4005: 'INT16_MIN': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt.cpp(321): note: see previous definition of 'INT16_MIN'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(51): warning C4005: 'INT32_MIN': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt.cpp(324): note: see previous definition of 'INT32_MIN'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(53): warning C4005: 'INT8_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt.cpp(327): note: see previous definition of 'INT8_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(54): warning C4005: 'INT16_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt.cpp(330): note: see previous definition of 'INT16_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(55): warning C4005: 'INT32_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt.cpp(333): note: see previous definition of 'INT32_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(57): warning C4005: 'UINT8_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt.cpp(336): note: see previous definition of 'UINT8_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(58): warning C4005: 'UINT16_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt.cpp(339): note: see previous definition of 'UINT16_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(59): warning C4005: 'UINT32_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt.cpp(342): note: see previous definition of 'UINT32_MAX'
[build] [108/124  79% :: 19.369] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\kit\kitSop.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [109/124  80% :: 19.433] Building CXX object _deps\stp-build\lib\Parser\CMakeFiles\parser.dir\parsesmt2.tab.cpp.obj
[build] [110/124  81% :: 19.447] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\kit\kitIsop.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] [111/124  82% :: 19.471] Building C object _deps\stp-build\lib\extlib-abc\CMakeFiles\abc.dir\aig\kit\kitTruth.c.obj
[build] cl : Command line warning D9025 : overriding '/W3' with '/W1'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(49): warning C4005: 'INT8_MIN': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt2.cpp(318): note: see previous definition of 'INT8_MIN'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(50): warning C4005: 'INT16_MIN': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt2.cpp(321): note: see previous definition of 'INT16_MIN'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(51): warning C4005: 'INT32_MIN': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt2.cpp(324): note: see previous definition of 'INT32_MIN'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(53): warning C4005: 'INT8_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt2.cpp(327): note: see previous definition of 'INT8_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(54): warning C4005: 'INT16_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt2.cpp(330): note: see previous definition of 'INT16_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(55): warning C4005: 'INT32_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt2.cpp(333): note: see previous definition of 'INT32_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(57): warning C4005: 'UINT8_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt2.cpp(336): note: see previous definition of 'UINT8_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(58): warning C4005: 'UINT16_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt2.cpp(339): note: see previous definition of 'UINT16_MAX'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\stdint.h(59): warning C4005: 'UINT32_MAX': macro redefinition
[build] D:\Work\FortifyIQ-win-Ninja-Debug\_deps\stp-build\lib\Parser\lexsmt2.cpp(342): note: see previous definition of 'UINT32_MAX'
```
  
</details>

This is because you can't just use `add_compile_options` to set warnings level on MSVC. You should replace the default warning level `/W3`. See [this](https://gitlab.kitware.com/cmake/cmake/-/issues/19084) issue for details.

Also you are using `add_compile_options` which does this globally. I would suggest to use `target_compile_options`.
But as for the level of warnings - I would suggest moving them to CI. Because it is otherwise impossible to disable warnings in your project when using `FetchContent` without patching your project. If you like the idea - I will send a PR.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -330,8 +330,8 @@ if (NOT MSVC)
     add_cxx_flag_if_supported("-Winit-self")
     add_cxx_flag_if_supported("-Wparentheses")
     add_cxx_flag_if_supported("-Wunreachable-code")
-    add_compile_options("-fPIC")
     # add_cxx_flag_if_supported("-flto") # slow compile and not enough benefits
+    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 endif()
 
 add_definitions("-D__STDC_LIMIT_MACROS")

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,13 +18,7 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
-project(STP)
 cmake_minimum_required(VERSION 3.3 FATAL_ERROR)
-set(CMAKE_C_STANDARD 99)
-set(CMAKE_CXX_STANDARD 14)
-
-include (GenerateExportHeader)
-include(GNUInstallDirs)
 
 if(POLICY CMP0048)
     #policy for VERSION in cmake 3.0
@@ -35,6 +29,24 @@ if(POLICY CMP0074)
     #policy for <PackageName>_ROOT variables
     cmake_policy(SET CMP0074 NEW)
 endif()
+
+if(POLICY CMP0092)
+    # Disable passing /W3 by default on MSVC
+    cmake_policy(SET CMP0092 NEW)
+endif()
+
+if(POLICY CMP0077)
+    # Allow to override options via regular variables
+    cmake_policy(SET CMP0077 NEW)
+endif()
+
+project(STP)
+
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_CXX_STANDARD 14)
+
+include(GenerateExportHeader)
+include(GNUInstallDirs)
 
 # Search paths for custom CMake modules
 set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/modules)
@@ -181,6 +193,8 @@ else ()
     endif()
 endif()
 
+option(SUPPRESS_WARNINGS "Silence all warnings" OFF)
+
 if (NOT MSVC)
     add_compile_options( -g)
     add_compile_options( -pthread )
@@ -236,7 +250,11 @@ else()
     add_compile_options(/GS)
 
     # Proper warning level
-    add_compile_options(/W1)
+    if(SUPPRESS_WARNINGS)
+        add_compile_options(/W0)
+    else()
+        add_compile_options(/W1)
+    endif()
 
     # Disable STL used in DLL-boundary warning
     add_compile_options(/wd4251)
@@ -304,32 +322,37 @@ if(APPLE)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
 endif()
 
-if (NOT MSVC)
-    add_cxx_flag_if_supported("-Wall")
-    add_cxx_flag_if_supported("-Wextra")
-    add_cxx_flag_if_supported("-Wunused")
-    add_cxx_flag_if_supported("-pedantic")
-    add_cxx_flag_if_supported("-Wsign-compare")
+if(NOT MSVC)
+    if(SUPPRESS_WARNINGS)
+        add_cxx_flag_if_supported("-w")
+    else()
+        add_cxx_flag_if_supported("-Wall")
+        add_cxx_flag_if_supported("-Wextra")
+        add_cxx_flag_if_supported("-Wunused")
+        add_cxx_flag_if_supported("-pedantic")
+        add_cxx_flag_if_supported("-Wsign-compare")
+        add_cxx_flag_if_supported("-Wtype-limits")
+        add_cxx_flag_if_supported("-Wuninitialized")
+        add_cxx_flag_if_supported("-Wno-deprecated")
+        add_cxx_flag_if_supported("-Wstrict-aliasing")
+        add_cxx_flag_if_supported("-Wpointer-arith")
+        add_cxx_flag_if_supported("-Wheader-guard")
+        add_cxx_flag_if_supported("-Wpointer-arith")
+        add_cxx_flag_if_supported("-Wformat-nonliteral")
+        add_cxx_flag_if_supported("-Winit-self")
+        add_cxx_flag_if_supported("-Wparentheses")
+        add_cxx_flag_if_supported("-Wunreachable-code")
+    endif()
+
     if (NOT CMAKE_BUILD_TYPE STREQUAL "Release")
         add_cxx_flag_if_supported("-fno-omit-frame-pointer")
     endif()
-    add_cxx_flag_if_supported("-Wtype-limits")
-    add_cxx_flag_if_supported("-Wuninitialized")
-    add_cxx_flag_if_supported("-Wno-deprecated")
-    add_cxx_flag_if_supported("-Wstrict-aliasing")
-    add_cxx_flag_if_supported("-Wpointer-arith")
-    add_cxx_flag_if_supported("-Wheader-guard")
     #if(NOT ENABLE_TESTING AND ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
         # almost working
         # add_cxx_flag_if_supported("-fvisibility=hidden")
         set(CMAKE_EXE_LINKER_FLAGS " ${CMAKE_EXE_LINKER_FLAGS} -Wl,--discard-all -Wl,--build-id=sha1")
     endif()
-    add_cxx_flag_if_supported("-Wpointer-arith")
-    add_cxx_flag_if_supported("-Wformat-nonliteral")
-    add_cxx_flag_if_supported("-Winit-self")
-    add_cxx_flag_if_supported("-Wparentheses")
-    add_cxx_flag_if_supported("-Wunreachable-code")
     # add_cxx_flag_if_supported("-flto") # slow compile and not enough benefits
     set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 endif()

--- a/lib/Sat/CryptoMinisat5.cpp
+++ b/lib/Sat/CryptoMinisat5.cpp
@@ -169,7 +169,7 @@ uint32_t CryptoMiniSat5::getFixedCountWithAssumptions(const stp::SATSolver::vec_
 
   //std::cerr << assumps.size() << " assumptions" << std::endl;
 
-  unsigned assigned = 0;
+  uint32_t assigned = 0;
   std::vector<CMSat::Lit> zero = s->get_zero_assigned_lits();
   for (CMSat::Lit l : zero)
   {
@@ -183,7 +183,7 @@ uint32_t CryptoMiniSat5::getFixedCountWithAssumptions(const stp::SATSolver::vec_
 
   // The assumptions are each single literals (corresponding to bits) that are true/false. 
   // so in the result they should be all be set 
-  assert(assigned >= assumps.size());
+  assert(assigned >= static_cast<uint32_t>(assumps.size()));
   assert(s->get_sum_conflicts() == conf ); // no searching, so no conflicts.
   assert(CMSat::l_False != r); // always satisfiable.
 

--- a/lib/Simplifier/DifficultyScore.cpp
+++ b/lib/Simplifier/DifficultyScore.cpp
@@ -54,7 +54,7 @@ long eval(const ASTNode& b)
     const auto cbv = b[0].GetBVConst(); // cleanup?
     bool last = CONSTANTBV::BitVector_bit_test(cbv,0);
     int changes = 0;
-    for (int i =1; i < b.GetValueWidth();i++)
+    for (unsigned int i =1; i < b.GetValueWidth();i++)
     {
         if (last != CONSTANTBV::BitVector_bit_test(cbv,i))
           changes++;

